glacierscript.py

```diff
@@ -706,21 +709,84 @@ def withdraw_interactive():
         print "\n\n*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***\n"
 
         source_address = raw_input("\nSource cold storage address: ")
-        addresses[source_address] = 0
 
-        redeem_script = raw_input("\nRedemption script for source cold storage address: ")
+        if RE_SIGN_MODE is not 1:
+            redeem_script = raw_input("\nRedemption script for source cold storage address: ")
+            dest_address = raw_input("\nDestination address: ")
+            num_tx = int(raw_input("\nHow many unspent transactions will you be using for this withdrawal? "))
+        else:
+            # begin main re-sign code block
+            print "\nPlease paste the partially-signed raw transaction (hexadecimal format) with unspent outputs at the source address"
+            print "OR"
+            print "input a filename located in the current directory which contains the raw transaction data"
+            print "(If the transaction data is over ~4000 characters long, you _must_ use a file.):"
 
-        dest_address = raw_input("\nDestination address: ")
-        addresses[dest_address] = 0
+            part_signed_hex_tx = raw_input()
+            if os.path.isfile(part_signed_hex_tx):
+                part_signed_hex_tx = open(part_signed_hex_tx).read().strip()
+
+            part_signed_tx = json.loads(bitcoin_cli_call("decoderawtransaction",part_signed_hex_tx))
+            redeem_script=part_signed_tx["vin"][0]["txinwitness"][-1]
```

Comment:
Will this work if the Glacier cold storage address is an older, pre-segwit address? It's important that Glacier continue to fully work for all previously-created addresses.

Reply:
Glacier being designed for non-technical users, many people using it might have no idea if their cold storage address is segwit or not.

In the developer tests, there is a withdrawal using non-segwit. You could use those keys to test the non-segwit sequential signing. I suspect the code required to handle it will be simple.